app/src/utils/walletStore.ts

```diff
@@ -83,13 +83,56 @@ export const walletStore = writable<WalletStore>({
 	signMessage: undefined
 });
 
-export const walletNameStore = writable<WalletNameStore>({
-	walletName: null
-});
+function createWalletNameStore() {
+	const { subscribe, set, update } = writable<WalletNameStore>({
+		walletName: null
+	});
+
+	return {
+		subscribe,
+		updateName: (walletName: WalletName) => update(() => ({ walletName })),
```

Comment:
Since this store has only one value, we can use `set()` directly to replace the new value of `walletName`.

Reply:
Nice.